### PR TITLE
Test/course

### PR DESCRIPTION
**Pull request by nugentmichael1**
- Unit tests
- More routes.  All but two tested.
- Swagger dependency
- Documentation

### CODE REVIEW

**Comment by treycrossley** [comment]
Could you separate these Pull Requests into 4 separate ones?
1) Unit Tests
2) Routes
3) Swagger Dependency
4) Documentation

Just so if there's an error with one of them, we can roll it back without it removing all the other dope functionality you implemented

**Comment by treycrossley** [comment]
That's crazy that you already implemented swagger tho, hella fast!! good shit!

File: .env [hunk 1]
```diff
@@ -0,0 +1,3 @@
+SPRING_DATASOURCE_USERNAME=p2team1_user
```
Comment (treycrossley) [inline .env:1]: .env shouldn't be committed to into version control. Security issue to have passwords for the entire world to see

File: pom.xml [hunk 2]
```diff
@@ -1,84 +1,110 @@
...
+		<developer/>
+	</developers>
+	<scm>
+		<connection/>
+		<developerConnection/>
+		<tag/>
+		<url/>
+	</scm>
+	<properties>
+		<java.version>17</java.version>
```
Comment (treycrossley) [inline pom.xml:30]: I think this is the old version of the pom.xml? can you merge main

**Review by treycrossley — CHANGES_REQUESTED** [review]
at first glance, it looks good! I'll do some more in depth testing once the PRs are split up

File: pom.xml (hunk 2 above)
Comment (nugentmichael1) [reply pom.xml:30]: This was merged from main.

Comment (treycrossley) [reply pom.xml:30]: the pom doesn't match the version currently in main? did you add any extra dependencies

Comment (nugentmichael1) [reply pom.xml:30]: There are four more dependencies: Swagger, JUnit, Mockito, and security.  You can look at commit history too to see when someone has merged main into their branch.

**Comment by nugentmichael1** [comment]
I don't think I can cleanly separate the branch into four different branches because I did a little bit of the work on each intermittently and the commit history is messy.

**Pull request closed without merging**